lib/marcel/magic.rb

```diff
@@ -119,7 +119,7 @@ def self.magic_match(io, method)
 
       io.binmode if io.respond_to?(:binmode)
       io.set_encoding(Encoding::BINARY) if io.respond_to?(:set_encoding)
-      buffer = "".encode(Encoding::BINARY)
+      buffer = (+"").encode(Encoding::BINARY)
```

Comment:
Perhaps this is better for readability: 

```suggestion
      buffer = String.new.encode(Encoding::BINARY)
```

Reply:
I see from the bench-marking that you did already consider that option. Also this is more succinct hence better.

```suggestion
      buffer = String.new(encoding: Encoding::BINARY)
```

Reply:
Please disregard!

Reply:
`String.new.encode(Encoding::BINARY)` looks good to me.

Reply:
I think what you want here is to declare a mutable string with binary encoding. That's

```
String.new(encoding: Encoding::BINARY)
```